src/popover/src/Popover.js

```diff
@@ -1,64 +1,184 @@
+
+    /**
+     * Function called when the Popover opens.
+     */
+    onOpen: PropTypes.func.isRequired,
+
+    /**
+     * Function fired when Popover closes.
+     */
+    onClose: PropTypes.func.isRequired,
+
+    /**
+     * Function that will be called when the enter transition is complete.
+     */
+    onOpenComplete: PropTypes.func.isRequired,
+
+    /**
+     * Function that will be called when the exit transition is complete.
+     */
+    onCloseComplete: PropTypes.func.isRequired,
+
+    /**
+     * When true, bring focus inside of the Popover on open.
+     */
+    bringFocusInside: PropTypes.bool
   }
 
   static defaultProps = {
-    side: 'bottom',
+    position: Position.BOTTOM,
+    isShown: false,
+    minWidth: 200,
+    minHeight: 40,
+    animationDuration: 300,
+    zIndex: 40,
     onOpen: () => {},
     onClose: () => {},
-    minWidth: 200,
-    minHeight: 40
+    onOpenComplete: () => {},
+    onCloseComplete: () => {},
+    bringFocusInside: true
   }
 
-  constructor() {
-    super()
+  constructor(props) {
+    super(props)
     this.state = {
-      isOpen: false,
-      targetRect: {}
-    }
-  }
-
-  componentWillReceiveProps(nextProps) {
-    if (Object.prototype.hasOwnProperty.call(nextProps, 'isOpen')) {
-      if (nextProps.isOpen) {
-        this.setState({
-          targetRect: this.getTargetRect()
-        })
-      }
+      isShown: props.isShown
     }
   }
 
   componentWillUnmount() {
     document.body.removeEventListener('click', this.onBodyClick, false)
     document.body.removeEventListener('keydown', this.onEsc, false)
-    window.removeEventListener('resize', this.resize, false)
+  }
+
+  /**
+   * Methods borrowed from BlueprintJS
+   * https://github.com/palantir/blueprint/blob/release/2.0.0/packages/core/src/components/overlay/overlay.tsx
+   */
+  bringFocusInside = () => {
+    if (!this.props.bringFocusInside) return
+
+    // Always delay focus manipulation to just before repaint to prevent scroll jumping
+    return requestAnimationFrame(() => {
+      // Container ref may be undefined between component mounting and Portal rendering
+      // activeElement may be undefined in some rare cases in IE
+
+      if (
+        this.popoverNode == null || // eslint-disable-line eqeqeq, no-eq-null
+        document.activeElement == null || // eslint-disable-line eqeqeq, no-eq-null
+        !this.props.isShown
+      ) {
+        return
+      }
+
+      const isFocusOutsideModal = !this.popoverNode.contains(
+        document.activeElement
+      )
+      if (isFocusOutsideModal) {
+        // Element marked autofocus has higher priority than the other clowns
+        const autofocusElement = this.popoverNode.querySelector('[autofocus]')
+        const wrapperElement = this.popoverNode.querySelector('[tabindex]')
+        const buttonElement = this.popoverNode.querySelector('button')
+
+        // eslint-disable-next-line eqeqeq, no-eq-null
+        if (autofocusElement != null) {
```

Comment:
You should use `if (autofocusElement) {` and remove all the `eslint-disable-next-line`'s.